--- a/deps/v8/src/codegen/code-stub-assembler.h
+++ b/deps/v8/src/codegen/code-stub-assembler.h
@@ -6,6 +6,9 @@
 #define V8_CODEGEN_CODE_STUB_ASSEMBLER_H_
 
 #include <functional>
+#include "src/objects/cell.h"
+#include "src/objects/js-generator.h"
+#include "src/objects/js-promise.h"
 
 #include "src/base/macros.h"
 #include "src/codegen/bailout-reason.h"
--- a/deps/v8/src/codegen/tnode.h
+++ b/deps/v8/src/codegen/tnode.h
@@ -247,7 +247,7 @@
 struct is_subtype {
   static const bool value =
       std::is_base_of<U, T>::value || (std::is_same<U, MaybeObject>::value &&
-                                       std::is_convertible<T, Object>::value);
+                                       true);
 };
 template <class T1, class T2, class U>
 struct is_subtype<UnionT<T1, T2>, U> {
